lib/elasticity/index_config.rb

```diff
@@ -20,7 +22,15 @@ def client
     end
 
     def definition
-      { settings: @elasticity_config.settings, mappings: { @document_type => @mapping } }
+      definition = { settings: @elasticity_config.settings, mappings: { @document_type => @mapping || {} } }
+      subclasses.each do |doc_type, subclass|
+        definition[:mappings][doc_type] = subclass.constantize.mapping
+      end if subclasses.present?
+      definition
+    end
+
+    def document_types
+      definition[:mappings].collect { |doc_type, mapping| doc_type }
```

Comment:
We probably want to memoize these two methods so that calling it multiple times won't recalculate everything